pkg/k8shandler/collection.go

```diff
@@ -366,3 +368,23 @@ func isBufferFlushRequired(current *apps.DaemonSet, desired *apps.DaemonSet) boo
 func getServiceAccountLogCollectorUID() types.UID {
 	return serviceAccountLogCollectorUID
 }
+
+func envValueEqual(cenv, denv []core.EnvVar) bool {
```

Comment:
I don't think this handles the case where the user may have added or deleted env vars to the daemonset.  I think you'll have to add logic to say "if celem not in denv then return false" and "if delem not in cenv then return false"

Reply:
:p indeed...  thanks!